### PR TITLE
added methods to replace invalid azure symbols

### PR DESCRIPTION
**Pull request by tarurar**

### CODE REVIEW

File: src/Lykke.Common/StringUtils.cs [hunk 1]
```diff
@@ -578,9 +580,24 @@ public static bool IsValidPartitionOrRowKey(this string src)
             if (string.IsNullOrWhiteSpace(src))
                 return false;
             
-            return !Regex.IsMatch(src, @"[\p{C}|/|\\|#|?]");
+            return !Regex.IsMatch(src, AzureInvalidSymbolsPattern);
         }
-        
+
+        public static string RefinePartitionOrRowKey(this string src, string replaceStr = "_")
```
Comment (starkmsu) [inline src/Lykke.Common/StringUtils.cs]: Xml doc here and below?

File: src/Lykke.Common/StringUtils.cs [hunk 2]
```diff
@@ -578,9 +580,24 @@ public static bool IsValidPartitionOrRowKey(this string src)
...
+        {
+            return new Regex(AzureInvalidSymbolsPattern).Replace(src, replaceStr);
```
Comment (starkmsu) [inline src/Lykke.Common/StringUtils.cs]: Input params validation?

File: src/Lykke.Common/StringUtils.cs [hunk 3]
```diff
@@ -578,9 +580,24 @@ public static bool IsValidPartitionOrRowKey(this string src)
...
+        }
+
+        public static string EnsurePartitionOrRowKeyValid(this string src)
+        {
+            if (!src.IsValidPartitionOrRowKey())
```
Comment (starkmsu) [inline src/Lykke.Common/StringUtils.cs]: Input param validation?

File: src/Lykke.Common/StringUtils.cs [hunk 4]
```diff
@@ -573,14 +575,58 @@ public static string SanitizePhone(this string phone)
...
+            if (string.IsNullOrWhiteSpace(src))
+                throw new ArgumentNullException(nameof(src));
+
+            if (string.IsNullOrWhiteSpace(replaceStr))
+                throw new ArgumentNullException(nameof(replaceStr));
+
+            if (!replaceStr.IsValidPartitionOrRowKey())
+                throw new ArgumentException("String contains invalid symbols", nameof(replaceStr));
+
+            return new Regex(AzureInvalidSymbolsPattern).Replace(src, replaceStr);
```
Comment (starkmsu) [inline src/Lykke.Common/StringUtils.cs]: Need to vlidate that replaceStr doesn't contain symbols from AzureInvalidSymbolsPattern

Comment (tarurar) [reply src/Lykke.Common/StringUtils.cs]: It is already being done 3 lines before.

**Pull request merged**